### PR TITLE
Syndicate combat defib buffed, has unique sprites

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Combat defillibrator has halved recharge time. This allows it to both stun people more quickly, and actually be able to cause a heart attack with harm intent during the stun.
It also has unique, mediocre sprites by Me

image

Syndicate duffel bag sprites altered slightly so the little med/ammo icon is tilted 
image

## Why It's Good For The Game

I randomly found out combat defibs could actually be used offensively while testing things, but they were both a tad slow to recharge and had what was practically an undocumented feature that i've never seen anyone use. This PR points out that they can be used in combat, and buffs them appropiately so a nuclear operative could reasonably use it, at the cost of sacrificing their belt slot and their offhand.

## Changelog
:cl:
tweak: halved syndicate combat defib recharge time
imageadd: it also has new sprites
/:cl:

### CODE REVIEW

**Comment by Whoneedspacee** [comment]
this is the good shit we need

**Comment by Akrilla** [comment]
contract kit buff

File: code/game/objects/items/defib.dm [hunk 1]
```diff
@@ -273,6 +275,21 @@
...
 	update_icon()
 
+/obj/item/defibrillator/compact/combat/cooldowncheck(mob/user)
+	addtimer(CALLBACK(src, .proc/finish_charging), 2.5 SECONDS)
+
+/obj/item/defibrillator/compact/combat/make_paddles()
+	return new /obj/item/twohanded/shockpaddles/syndicate(src)
+
+/obj/item/twohanded/shockpaddles/syndicate/update_icon()
+	icon_state = "syndiepaddles[wielded]"
```
Comment (nemvar) [inline code/game/objects/items/defib.dm]: don't just copy/paste the same code over again. Just add a new variable to the paddles that defines the base icon_state

File: code/game/objects/items/defib.dm [hunk 2]
```diff
@@ -273,6 +275,21 @@
 	cell = new /obj/item/stock_parts/cell/infinite(src)
 	update_icon()
 
+/obj/item/defibrillator/compact/combat/cooldowncheck(mob/user)
+	addtimer(CALLBACK(src, .proc/finish_charging), 2.5 SECONDS)
```
Comment (nemvar) [inline code/game/objects/items/defib.dm]: Same here, just add a variable that changes cooldown time.

File: code/game/objects/items/defib.dm [hunk 3]
```diff
@@ -676,11 +693,14 @@
...
-	desc = "A pair of paddles used to revive deceased operatives. It possesses both the ability to penetrate armor and to deliver powerful shocks offensively."
+	desc = "A pair of paddles used to revive deceased operatives. It possesses both the ability to penetrate armor and to deliver powerful or disabling shocks offensively."
 	combat = TRUE
 	icon = 'icons/obj/defib.dmi'
-	icon_state = "defibpaddles0"
-	item_state = "defibpaddles0"
+	icon_state = "syndiepaddles0"
+	item_state = "syndiepaddles0"
+	req_defib = TRUE
+a
```
Comment (nemvar) [inline code/game/objects/items/defib.dm]: wat?

Comment (carlarctg) [reply code/game/objects/items/defib.dm]: i have no idea how an a got in there

Comment (Akrilla) [reply code/game/objects/items/defib.dm]: a

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
what nemvar said

**Comment by optimumtact** [comment]
power creep

**Comment by carlarctg** [comment]
ok done

pls merge :( if not ill remove the halved charge i GUESS

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/game/objects/items/defib.dm [hunk 4]
```diff
@@ -231,7 +231,10 @@
 			return FALSE
 
 /obj/item/defibrillator/proc/cooldowncheck(mob/user)
-	addtimer(CALLBACK(src, .proc/finish_charging), 5 SECONDS)
+	if(!obj/item/defibrillator/compact/combat)
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/game/objects/items/defib.dm]: done, if the lack of / is what you meant

File: code/game/objects/items/defib.dm [hunk 5]
```diff
@@ -231,7 +231,10 @@
...
+		addtimer(CALLBACK(src, .proc/finish_charging), 5 SECONDS)
+	else
+		addtimer(CALLBACK(src, .proc/finish_charging), 2.5 SECONDS)
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/game/objects/items/defib.dm]: i did that yay

**Comment by carlarctg** [comment]
apparently the conflict fixed itself

File: code/game/objects/items/defib.dm [hunk 6]
```diff
@@ -264,15 +265,21 @@
...
 	safety = FALSE
+	cooldown_duration = 2.5 SECONDS
 
 /obj/item/defibrillator/compact/combat/loaded/Initialize()
 	. = ..()
 	cell = new /obj/item/stock_parts/cell/infinite(src)
 	update_icon()
 
+/obj/item/defibrillator/compact/combat/make_paddles()
+	return new /obj/item/twohanded/shockpaddles/syndicate(src)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/defib.dm]: WHO DID THIS

**Comment by SpaceManiac** [comment]
Yes, you addressed what I was worried about, will try to give another once-over when free

**Comment by ExcessiveUseOfCobblestone** [comment]
bro convert that proc into a variable pleaseeeeee

**Comment by carlarctg** [comment]
re ok

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 7]
```diff
@@ -591,7 +591,7 @@
 	basic_modules = list(
 		/obj/item/assembly/flash/cyborg,
 		/obj/item/reagent_containers/borghypo/syndicate,
-		/obj/item/twohanded/shockpaddles/syndicate,
+		/obj/item/twohanded/shockpaddles/syndicate/cyborg,
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: document this.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 7 above)
Comment (carlarctg) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: it's literally the same thing as before, i just had to make it a subtype so the normal syndie paddles couldn't be dropped

**Comment by carlarctg** [comment]
@ExcessiveUseOfCobblestone done

**Comment by ExcessiveUseOfCobblestone** [comment]
bro u trolled me

File: code/game/objects/items/defib.dm [hunk 8]
```diff
@@ -192,7 +193,10 @@
 		A.UpdateButtonIcon()
 
 /obj/item/defibrillator/proc/make_paddles()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/defib.dm]: this proc is ABSOLUTELY USELESS.

on the defib
```DM
var/paddle_type = /your/fav/type/here
```

wherever makepaddles is used
```DM
var/blah = make_paddles() >>>>>> var/blah = new paddle_type(src)
```

Comment (carlarctg) [reply code/game/objects/items/defib.dm]: i don't know what these words mean cobby man i have not touched this part at all in the pr

Comment (carlarctg) [reply code/game/objects/items/defib.dm]: pls don't make me fix someone else's shitcode i can barely handle my own

**Review by SpaceManiac — APPROVED** [review]
patch is fine as of now, cobby can just fix that one organization thing themselves

**Comment by ExcessiveUseOfCobblestone** [comment]
cringe

**Comment by SpaceManiac** [comment]
fix it then

**Pull request merged**